src/Microsoft.VisualStudio.Services.Agent/HostContext.cs

```diff
@@ -28,7 +29,7 @@ public interface IHostContext : IDisposable, IKnobValueContext
         StartupType StartupType { get; set; }
         CancellationToken AgentShutdownToken { get; }
         ShutdownReason AgentShutdownReason { get; }
-        ISecretMasker SecretMasker { get; }
+        LoggedSecretMasker SecretMasker { get; }
```

Comment:
Why did the return type change? Wouldn't returning an interface be better here?

Reply:
Same in other places in this class

Reply:
Generally yes, it will be better. But since I am using not exactly the `ISecretMasker` interface, I have added a new one - `ILoggedSecretMasker`, which extends the behavior of `ISecretMaslker ` interface and I replaced all the entries of `ISecretMasker` to `ILoggedSecretMasker`.